cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -25,94 +29,40 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
 
 
   "A Cwl object for 1st-tool" should "convert to WOM" in {
-    val firstTool =
-      """
-cwlVersion: v1.0
-class: CommandLineTool
-baseCommand: echo
-inputs:
-- type: string
-  inputBinding:
-    position: 1
-  id: message
-outputs: []
-"""
-
-    CwlCodecs.decodeCwl(firstTool) map {
-      case (clt: CommandLineTool, _) =>
-        clt.womExecutable match {
-          case Valid(wom) =>
-            wom.entryPoint match {
-              case taskDefinition: TaskDefinition =>
-
-                taskDefinition.inputs shouldBe Set(RequiredInputDefinition("message", WdlStringType))
-
-                taskDefinition.graph.map {
-                  graph =>
-                    graph.nodes.collect { case gin: GraphInputNode => gin.name } should be(Set("echo.message"))
-                    graph.nodes collect { case cn: CallNode => cn.name } should be(Set("echo"))
-
-                    graph.nodes.collectFirst { case echo: CallNode if echo.name == "echo" => echo }.get.
-                      upstream shouldBe Set(RequiredGraphInputNode("echo.message", WdlStringType))
-                }
-
-              case _ => fail("not a task definition")
-            }
-          case i@Invalid(_) => fail(s"Invalid: $i")
-        }
-      case (wth: Any, _) => fail(s"parsed unexpected type $wth")
-    } leftMap { error => fail(s"did not parse!  $error") }
+    (for {
+      clt <- decodeAllCwl(rootPath/"1st-tool.cwl").
+              map(_.select[CommandLineTool].get).
+              value.
+              unsafeRunSync
+
+      wom <-  clt.womExecutable.toEither
+    } yield validateWom(wom)).leftMap(e => throw new RuntimeException(s"error! $e"))
+
+    def validateWom: Executable => Unit =
```

Comment:
I wasn't sure if the def "interned" the function or created a new one each time, but the val is definitely a one-time thing.  🙂